SonarQube.TeamBuild.PreProcessor/ProcessedArgs.cs

```diff
@@ -69,13 +69,25 @@ public ProcessedArgs(string key, string name, string version, IAnalysisPropertyP
 
         public IAnalysisPropertyProvider GlobalFileProperties { get { return this.globalFileProperties; } }
 
+        /// <summary>
+        /// Returns the value for the specified setting.
+        /// Throws if the setting does not exist.
+        /// </summary>
         public string GetSetting(string key)
         {
             string value;
-            this.aggProperties.TryGetValue(key, out value);
+            if (!this.aggProperties.TryGetValue(key, out value))
+            {
+                string message = string.Format(System.Globalization.CultureInfo.CurrentCulture, Resources.ERROR_MissingSetting, key);
+                throw new InvalidOperationException(message);
+            }
```

Comment:
I changed GetSetting to throw on missing settings and added a TryGetSetting that won't throw. It makes it clearer what we expect to happen in each case.